force-app/main/default/lwc/ldsCreateRecord/__tests__/ldsCreateRecord.test.js

```diff
@@ -0,0 +1,138 @@
+            { apiName: 'Account', fields: { Name: USER_INPUT } }
+        ];
+
+        const element = createElement('c-lds-create-record', {
+            is: LdsCreateRecord
+        });
+        document.body.appendChild(element);
+
+        const inputEl = element.shadowRoot.querySelector(
+            'lightning-input[class="slds-m-bottom_x-small"]'
+        );
+        inputEl.value = USER_INPUT;
+        inputEl.dispatchEvent(new CustomEvent('change'));
+
+        const buttonEl = element.shadowRoot.querySelector('lightning-button');
+        buttonEl.click();
+
+        // Return a promise to wait for any asynchronous DOM updates.
+        return Promise.resolve().then(() => {
+            expect(createRecord).toHaveBeenCalled();
+            expect(createRecord.mock.calls[0]).toEqual(INPUT_PARAMETERS);
+        });
+    });
+
+    it('displays an account id after record creation', () => {
+        const USER_INPUT = 'Gomez Inc.';
+        const RESULT = '0011700000r6zdbAAA';
+        createRecord.mockResolvedValue(mockCreateRecord);
+
+        const element = createElement('c-lds-create-record', {
+            is: LdsCreateRecord
+        });
+        document.body.appendChild(element);
+
+        const inputEl = element.shadowRoot.querySelector(
+            'lightning-input[class="slds-m-bottom_x-small"]'
+        );
+        inputEl.value = USER_INPUT;
+        inputEl.dispatchEvent(new CustomEvent('change'));
+
+        const buttonEl = element.shadowRoot.querySelector('lightning-button');
+        buttonEl.click();
+
+        // Return a promise to wait for any asynchronous DOM updates.
+        return flushPromises().then(() => {
+            const displayEl = element.shadowRoot.querySelector(
+                'lightning-input:not([class="slds-m-bottom_x-small"])'
+            );
+            expect(displayEl.value).toBe(RESULT);
+        });
+    });
+
+    it('displays a success toast after record creation', () => {
+        const USER_INPUT = 'Gomez Inc.';
+        createRecord.mockResolvedValue(mockCreateRecord);
+
+        const element = createElement('c-lds-create-record', {
+            is: LdsCreateRecord
+        });
+        document.body.appendChild(element);
+
+        // Mock handler for toast event
+        const handler = jest.fn();
+        // Add event listener to catch toast event
+        element.addEventListener('lightning__showtoast', handler);
+
+        const inputEl = element.shadowRoot.querySelector(
+            'lightning-input[class="slds-m-bottom_x-small"]'
+        );
+        inputEl.value = USER_INPUT;
+        inputEl.dispatchEvent(new CustomEvent('change'));
+
+        const buttonEl = element.shadowRoot.querySelector('lightning-button');
+        buttonEl.click();
+
+        // Return a promise to wait for any asynchronous DOM updates.
+        return Promise.resolve().then(() => {
+            // Check if toast event has been fired
+            expect(handler).toHaveBeenCalled();
```

Comment:
I can only verify if the event got fired. How can I verify the details (like if it was a `success` or `error` toast event)? @trevor-bliss @kevinv11n

Reply:
Similar to how we override the navigation mock to capture and return the input parameters, we can override the `platformShowToastEvent` mock to capture the params the event is initialized with. Then the test can verify retrieved the param from the mock and validate it was the expected "success" or "error" variant.

Reply:
Using a custom mock now, thanks!